pkg/execution/state/redis_state/redis_state.go

```diff
@@ -1013,44 +1049,47 @@ func (m unshardedMgr) DeletePause(ctx context.Context, p state.Pause) error {
 	}
 }
 
-func (m mgr) ConsumePause(ctx context.Context, pauseID uuid.UUID, data any) error {
+func (m mgr) ConsumePause(ctx context.Context, pauseID uuid.UUID, data any) (didConsume bool, hasPendingSteps bool, err error) {
 	p, err := m.unshardedMgr.PauseByID(ctx, pauseID)
 	if err != nil {
-		return err
+		return
 	}
 
-	err = m.shardedMgr.consumePause(ctx, p, data)
+	didConsume, hasPendingSteps, err = m.shardedMgr.consumePause(ctx, p, data)
 	if err != nil {
-		return err
+		return
 	}
 
 	// The pause was now consumed, so let's clean up
-	return m.unshardedMgr.DeletePause(ctx, *p)
+	err = m.unshardedMgr.DeletePause(ctx, *p)
+	return
 }
 
-func (m shardedMgr) consumePause(ctx context.Context, p *state.Pause, data any) error {
+func (m shardedMgr) consumePause(ctx context.Context, p *state.Pause, data any) (didConsume bool, hasPendingSteps bool, err error) {
```

Comment:
Using named return variables like this can be confusing, but I'm specifying them here so that the caller has a better idea what they are; seeing `bool, bool, err` as the return doesn't give  much info.

We could refactor to a small struct if that feels more appropriate.